app/src/main/java/de/westnordost/streetcomplete/osm/Surface.kt

```diff
@@ -1,12 +1,30 @@
 package de.westnordost.streetcomplete.osm
 
-val ANYTHING_UNPAVED = setOf(
+val SOFT_SURFACES = setOf("ground", "earth", "dirt", "grass", "sand", "mud", "ice", "salt", "snow", "woodchips")
+
+val ANYTHING_UNPAVED = SOFT_SURFACES + setOf(
     "unpaved", "compacted", "gravel", "fine_gravel", "pebblestone", "grass_paver",
-    "ground", "earth", "dirt", "grass", "sand", "mud", "ice", "salt", "snow", "woodchips"
 )
 
-val ANYTHING_PAVED = setOf(
+val ANYTHING_FULLY_PAVED = setOf(
     "paved", "asphalt", "cobblestone", "cobblestone:flattened", "sett",
-    "concrete", "concrete:lanes", "concrete:plates", "paving_stones",
+    "concrete", "concrete:plates", "paving_stones",
     "metal", "wood", "unhewn_cobblestone"
 )
+
+val ANYTHING_PAVED = ANYTHING_FULLY_PAVED + setOf(
+    "concrete:lanes"
+)
+
+fun isSurfaceAndTracktypeMismatching(surface: String, tracktype: String): Boolean {
+    if (tracktype == "grade1" && ANYTHING_UNPAVED.contains(surface)) {
+        return true
+    }
+    if (tracktype == "grade2" && SOFT_SURFACES.contains(surface)) {
+        return true
+    }
+    if ((tracktype == "grade3" || tracktype == "grade4" || tracktype == "grade5") && ANYTHING_FULLY_PAVED.contains(surface)) {
+        return true
+    }
```

Comment:
It should be possible to use `in` instead of `.contains()` because they are `Set`s:

```suggestion
    if (tracktype == "grade1" && surface in ANYTHING_UNPAVED) {
        return true
    }
    if (tracktype == "grade2" && surface in SOFT_SURFACES) {
        return true
    }
    if ((tracktype == "grade3" || tracktype == "grade4" || tracktype == "grade5") && surface in ANYTHING_FULLY_PAVED) {
        return true
    }
```